lib/FIR/statements.h

```diff
@@ -378,25 +401,26 @@ class LoadInsn : public MemoryStmt_impl {
 // Store value(s) from an applied expression to a location
 class StoreInsn : public MemoryStmt_impl {
 public:
-  using ValueType =
-      std::variant<Value, ApplyExprStmt *, Addressable_impl *, BasicBlock *>;
-  template<typename T> static StoreInsn Create(T *addr, T *value) {
-    return StoreInsn{addr, value};
-  }
-  template<typename T> static StoreInsn Create(T *addr, BasicBlock *value) {
+  using ValueType = std::variant<Value, QualifiedStmt<ApplyExprStmt>,
+      QualifiedStmt<Addressable_impl>, BasicBlock *>;
+
+  template<typename A>
+  static StoreInsn Create(QualifiedStmt<Addressable_impl> addr, A value) {
```

Comment:
shouldn't `A` also be tested with a QualifiefStmt<> ?

Reply:
If not, then is the QualifiedStmt <> test necessary here for `Addressable_impl` since we have that check in the constructors ?